packages/server-boilerplate/src/utils/useForwardUnhandledRequests.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+import { Express, NextFunction, Request, Response } from 'express';
+import { IncomingMessage, ServerResponse } from 'http';
+import { createProxyMiddleware, fixRequestBody } from 'http-proxy-middleware';
+import { UnauthenticatedError } from '@tupaia/utils';
+import { handleError } from './handleError';
+import { attachSession as defaultAttachSession } from '../orchestrator';
+
+const attachAuthorizationHeader = async (req: Request, res: Response, next: NextFunction) => {
+  const { session } = req;
+
+  if (!session) {
+    throw new UnauthenticatedError('Session is not attached');
+  }
+
+  req.headers.authorization = await session.getAuthHeader();
+
+  next();
+};
+
+type Middleware = (req: Request, res: Response, next: NextFunction) => void;
+
+export const useForwardUnhandledRequests = (
+  app: Express,
+  target: string,
+  prefix?: string,
+  attachSession: Middleware = defaultAttachSession,
+) => {
+  const options = {
+    target,
+    changeOrigin: true,
+    pathRewrite: (path: string) => {
```

Comment:
@chris-bes - I've extracted this function from admin-panel-server to server-boilerplate. I replaced app.path with path as it has the same value. Does that make sense to you?

Reply:
Yep I like it! Makes more sense that it sits in `server-boilerplate`